tools/serverpod_cli/bin/generator/class_analyzer.dart

```diff
@@ -659,3 +663,174 @@ class ClassAnalyzer {
     return true;
   }
 }
+
+extension on ClassAnalyzer {
```

Comment:
Why is this an extension, rather than a method of the ClassAnalyzer? Also, it looks like it just duplicates the code, which makes it harder to maintain as any changes would need to be added in multiple places. It would be better to modify the _analyzeClassFile to be able to parse the exceptions.